### PR TITLE
kafka: add more options to kafka operator + new healthcheck

### PR DESCRIPTION
**Pull request by zmalik**
This PR updates the kafka operator to version `0.1.1`

- add 100 more parameters to configure Kafka properties and set them to tuned defaults.
- update the docker image to `mesosphere/kafka:0.1.0-2.2.1` 
       --> use port check for the health check instead of connection to zk 
- added livenessprobe

### CODE REVIEW

File: repository/kafka/operator/params.yaml [hunk 1]
```diff
@@ -2,52 +2,567 @@ BROKER_COUNT:
...
+LOG_FLUSH_SCHEDULER_INTERVAL_MS:
+  default: "9223372036854775807"
+  description: "The frequency in ms that the log flusher checks whether any log needs to be flushed to disk"
+
+LOG_FLUSH_START_OFFSET_CHECKPOINT_INTERVAL_MS:
+  default: "60000"
+  description: "The frequency with which we update the persistent record of log start offset"
+
+LOG_RETENTION_BYTES:
+  displayName: "log.retention.bytes"
```
Comment (alenkacz) [inline repository/kafka/operator/params.yaml:101]: if display name is meant as really human readable one I would probably go with "Log retention in bytes" or "Maximum size of the log in bytes" something like that. Or not doing display names at all if you want - they are not used anywhere and right now I am not sure when will they be used. Up to you :)

**Review by alenkacz — APPROVED** [review]
LGTM! It sparked a lot of questions on the KUDO side as we discussed over slack but looking good 👍

File: repository/kafka/operator/params.yaml (hunk 1 above)
Comment (zmalik) [reply repository/kafka/operator/params.yaml:101]: I'm using `displayName` to capture exact same name as used in broker configs https://kafka.apache.org/documentation/#brokerconfigs 

So users can identify how to configure a broker using these params

File: repository/kafka/operator/templates/configmap.yaml [hunk 2]
```diff
@@ -87,15 +56,14 @@ data:
     # There are a few important trade-offs here:
     #    1. Durability: Unflushed data may be lost if you are not using replication.
     #    2. Latency: Very large flush intervals may lead to latency spikes when the flush does occur as there will be a lot of data to flush.
-    #    3. Throughput: The flush is generally the most expensive operation, and a small flush interval may lead to excessive seeks.
+    #    3. Throughput: The flush is generally the most expensive operation, and a small flush interval may lead to exceessive seeks.
```
Comment (porridge) [inline repository/kafka/operator/templates/configmap.yaml]: `excessive` is the correct spelling.

File: repository/kafka/operator/templates/statefulset.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ spec:
       containers:
         - name: k8skafka
           imagePullPolicy: Always
-          image: mesosphere/kafka:0.1-2.2.1
+          image: mesosphere/kafka:0.1.0-2.2.1
```
Comment (porridge) [inline repository/kafka/operator/templates/statefulset.yaml:26]: Is the version here supposed to be independent from the operator version?

**Review by porridge — CHANGES_REQUESTED** [review]

File: repository/kafka/operator/templates/statefulset.yaml (hunk 3 above)
Comment (zmalik) [reply repository/kafka/operator/templates/statefulset.yaml:26]: yes, `0.1.0` isn't related to operator version

File: repository/kafka/operator/templates/configmap.yaml (hunk 2 above)
Comment (zmalik) [reply repository/kafka/operator/templates/configmap.yaml]: good catch! thank you!

**Pull request merged**